tests/e2e/scale/noobaa/test_scale_obc_creation.py

```diff
@@ -0,0 +1,184 @@
+import logging
+import pytest
+import time
+
+from ocs_ci.ocs import constants, scale_noobaa_lib
+from ocs_ci.framework.testlib import scale, E2ETest
+from ocs_ci.ocs.resources.objectconfigfile import ObjectConfFile
+from ocs_ci.framework.pytest_customization.marks import (
+    vsphere_platform_required,
+    skipif_aws_creds_are_missing,
+)
+
+log = logging.getLogger(__name__)
+
+
+@scale
+class TestScaleOCBCreation(E2ETest):
+    """
+    OBC scale creation, creating up to max support number of OBCs.
+    OBCs are created in the Multicloud Object Gateway and
+    Ceph Object Gateway (RGW)
+
+    """
+
+    namespace = constants.OPENSHIFT_STORAGE_NAMESPACE
+    sc_name = constants.NOOBAA_SC
+    sc_rgw_name = constants.DEFAULT_STORAGECLASS_RGW
+    scale_obc_count = 1000
+    num_obc_batch = 50
+
+    @skipif_aws_creds_are_missing
```

Comment:
Does it mean that this will work on AWS cloud platform only? Afaik there is ongoing effort to make sure noobaa test cases work on all cloud platforms.

Reply:
The way I see it, there is no reason for the test to be limited to any one platform - cloud or on-prem.
Even if it needed other cloud providers, we already have working accounts for AWS, GCP, and Azure.
The IBM COS infra is in place, we're waiting for credentials

Reply:
I agree with @MeridianExplorer. Multi-cloud tests shouldn't be limited to only one platform.
That said, we should cover more cloud providers and don't focus only on aws (we can parametrize tests with cloud providers or create new test cases for other providers).

Reply:
The way the test is written right now, the created OBCs will use the default backingstore. So if the cluster runs on Azure, it'll create N buckets that all lead to one Azure container. If the cluster runs on GCP, the test will create N buckets that all lead to one Google bucket, etc.

Either way, `bucket_factory()` takes care of it all automagically. As Filip said - it's just about passing the right parameters to the fixture.

Reply:
@MeridianExplorer @fbalak, For scale testing, we are experiencing memory issues when using default factory. 
When creating a large amount of PVCs/OBCs, there are many objects are created and they consume all memory on Jenkins slave which causes the test failing.  This issue you don't see in normal testing with small number of pvc/obc creation.
The recent incident that happened with PVC creation https://github.com/red-hat-storage/ocs-ci/issues/2744.
After multiple discussions, we decide to change the test case implementation by using kube_job instead.  We have seen the big improvement in terms of memory usage and shorten the time of PVC creation as well.

Reply:
@mbukatov I will remove aws creds check since we are supporting other cloud platforms.

Reply:
@tiffanyn108 - I understand, thanks for the elaboration.
Was `bucket_factory()` tested in scale, or was the decision based solely on PVC creation?
While I can understand why PVC objects can consume a lot of memory (being dependent on containing OCS and OCP objects within them) - OB was actually designed to be [extremely lightweight](https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/resources/objectbucket.py#L132-L141), and only contains 5 short strings per object (unlike the heavy-duty OBC which would not be needed/used for those tests)

Reply:
@MeridianExplorer I have done a comparison of memory usage between creating 1000 obcs using `bucket_factory()` and using `kube_job`.  Here is what I saw:
* `bucket_factory()` - Memory usage is ~2.2G - 2.5G 
* `kube_job` - Memory usage is ~500MB
Base on my observation, we can save ~5 times of memory used with kube_job implementation.  
In addition, with kube_job, the test execution time is 20mins faster.  I would refer to use kube_job for scale testing, especially with all the loads and large number of bucket creation.

Reply:
That's a very interesting and important benchmark. Thanks, Tifanny!